src/snowflake/connector/network.py

```diff
@@ -268,8 +268,16 @@ def get_connection(
                 )
             proxy_manager = self.proxy_manager_for(proxy)
 
-            # Add Host to proxy header SNOW-232777
-            proxy_manager.proxy_headers["Host"] = parsed_url.hostname
+            try:
+                # Add Host to proxy header SNOW-232777
+                proxy_manager.proxy_headers["Host"] = parsed_url.hostname
+            except AttributeError:
+                # log that the type proxy_manager not having attribute proxy_headers
+                logger.debug(
+                    f"Unable to set 'Host' to proxy manager of type {type(proxy_manager)} as"
+                    f" it does not have attribute 'proxy_headers'."
+                )
```

Comment:
do you think `logger.warning` is a better log level for this? how common is this issue?

Reply:
so far it's only happening to socks5 proxy, and depending on the issue amount (only 1 in our repo). I don't think it's quite common.

depending on user description `I modified the code to not set the header when using a SOCKS5 proxy, which seems to work. The connection succeeds over the SOCKS5 proxy. It also works if I set the headers property instead of the proxy_headers property.` in the [issue](https://github.com/snowflakedb/snowflake-connector-python/issues/1550), not setting the proxy_header in socks5 proxy can also work.

so I treat this as an expected case and use the debug level.

but I'm open to update it to warning as we are coming from throwing error -- if this is a warning level message I will rephrase it as ` f"Unable to set 'Host' to proxy manager of type {type(proxy_manager)} as, it does not have attribute 'proxy_headers', proxy might not work as expected."`

Reply:
sounds good.